### PR TITLE
Closes #1339: Add scroll behavior support for full screen pages on SystemEngineView.

### LINKED ISSUE #1339: Fullscreen pages can be scrolled (because height of toolbar can be scrolled offscreen)
Our reference browser and sample browser have the same issue that is described in this Focus issue:
https://github.com/mozilla-mobile/focus-android/issues/3815

Let's investigate if we can find a solution for that.

### PR DESCRIPTION
**Pull request by Amejia481**

### CODE REVIEW

File: components/feature/scroll/src/main/java/mozilla/components/feature/scroll/ScrollFeature.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import mozilla.components.browser.session.SelectionAwareSessionObserver
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.EngineView
+
+/**
+ * Feature implementation for connecting an [EngineView] with any View that you want to coordinate scrolling
+ * behavior with.
+ */
+class ScrollFeature(
```
Comment (pocmo) [inline components/feature/scroll/src/main/java/mozilla/components/feature/scroll/ScrollFeature.kt]: naming nit: `Scroll` is not precise enough yet, I think. Can we find something that is describing better was that feature does? I was thinking of something like "ToolbarHidingFeature" but its more generic than just about the toolbar (however we should mention the toolbar use case in the comment maybe!).

Comment (pocmo) [reply components/feature/scroll/src/main/java/mozilla/components/feature/scroll/ScrollFeature.kt]: CoordinateScrollingFeature? SyncScrollingFeature? ScrollFlagsFeature?

Comment (Amejia481) [reply components/feature/scroll/src/main/java/mozilla/components/feature/scroll/ScrollFeature.kt]: Yeah name is hard 😅. I like `CoordinateScrollingFeature` 👍 .

File: README.md [hunk 2]
```diff
@@ -114,6 +114,8 @@ _Combined components to implement feature-specific use cases._
 
 * 🔴 [**Toolbar**](components/feature/toolbar/README.md) - A component that connects a (concept) toolbar implementation with the browser session module.
 
+* 🔴 [**Scroll**](components/feature/scroll/README.md) - A component coordinates scroll behavior with an `EngineView`.
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply README.md]: Fair enough! 👍

**Pull request merged**